numba/parfor.py

```diff
@@ -1657,13 +1657,26 @@ def _convert_loop(self, blocks):
                     args = inst.value.args
                     loop_kind, loop_replacing = self._get_loop_kind(inst.value.func.name,
                                                                     call_table)
+                    # Get the body of the header of the loops minus the branch terminator
+                    # The general approach is to prepend the header block to the first
+                    # body block and then let dead code removal handle removing unneeded
+                    # statements.  Not all statements in the header block are unnecessary.
+                    header_body = blocks[loop.header].body[:-1]
                     # find loop index variable (pair_first in header block)
-                    for stmt in blocks[loop.header].body:
+                    loop_index = None
+                    for hbi, stmt in enumerate(header_body):
                         if (isinstance(stmt, ir.Assign)
                                 and isinstance(stmt.value, ir.Expr)
                                 and stmt.value.op == 'pair_first'):
                             loop_index = stmt.target.name
+                            li_index = hbi
                             break
+                    assert(loop_index is not None)
+                    # Remove pair_first from header.
+                    # We have to remove the pair_first by hand since it causes problems
+                    # for some code below if we don't.
+                    header_body = header_body[:li_index] + header_body[li_index+1:]
```

Comment:
would it be a good idea to check that the thing being sliced out is actually a `pair_first`?

Reply:
The code above this assures that that is indeed the case.  We search for pair_first and loop_index gets set only if it is found then after the loop we assert it is found.  Thus, li_index guaranteed to be that of a pair_first.

Reply:
Ah I see, thanks.